iceprod/core/i3exec.py

```diff
@@ -341,6 +347,12 @@ def runner(config, url, debug=False, offline=False):
             args['cfgfile'] = os.path.join(os.getcwd(),args['cfgfile'])
         else:
             args['cfgfile'] = None
+    if args['cfgfile']:
+        cfgfile = load_config(args['cfgfile'])
+        task = args.pop('task')
+        if task:
+            cfgfile['options']['task'] = task
+        args['cfgfile'] = cfgfile
 
     # start iceprod
     main(**args)
```

Comment:
What if `args['cfgfile']` is None?  Then `args['task']` still exists, and main will raise an exception for a bad argument.